### PR TITLE
adds new acid spit genetics mutation obtainable through polysmorphs

### PR DESCRIPTION
**Pull request by iloveloopers**
# Document the changes in your pull request

adds the new acid spit mutation obtainable only from polysmorphs, similarly to fire breath
lets you spit acid that will slowly corrode objects and clothing on players
its weak but annoying
instabillity of 50
cooldown of 18 seconds

# Why is this good for the game?
more species specific mutations is good
the mutation is really only useful for melting stuff, its impractical to use it in combat unless your enemy is naked

# Testing

tested it, works. destroys a common object in 2 shots and clothing depends on the acid armor.

# Wiki Documentation

mutation name: Acid Spit
description: "An ancient mutation from xenomorphs that forces the salivary glands to produce acid"
indicators: none
message: "Your saliva burns your mouth!"
how to obtain: polysmorph species
imcompatible with: none
instabillity: 50
compatible chromosomes: reinforcement, stabilizier

# Changelog

:cl:  
rscadd: New Acid Spit mutation obtainable from Polysmorphs
soundadd: New Acid Spit sound effect
/:cl:

### CODE REVIEW

**Comment by neeshacark** [comment]
instability seems a bit high for what this does tbh, isnt fire breath lower?

**Comment by ju45he** [comment]
> instability seems a bit high for what this does tbh, isnt fire breath lower?

You could use this for SO fucking much. Fire breath lights em up a bit but fire extinguishers make it obsolete. Spitting corrosive acid at someone would eventually make their uniform corrode, anything in pockets belt slot ID allow are all for the taking with simple timing and with low risk should you simply use something to outrun them for a bit while you wait for their stuff to corrode.

**Comment by Tenn0x** [comment]
seems a little expensive but stabilizer can be used so idk

File: code/modules/projectiles/projectile/special/neurotoxin.dm [hunk 1]
```diff
@@ -12,3 +12,21 @@
...
+	range = 7
+	speed = 1.8 // spit is not very fast
+
+obj/item/projectile/bullet/acid/on_hit(atom/target, blocked = FALSE)
+	if(isalien(target)) // shouldn't work on xenos
+		nodamage = TRUE
+	else if(isobj(target) || isclosedturf(target))
+		target.acid_act(acidpwr = 50, acid_volume = 25) // does good damage to objects and structures
+	else if(iscarbon(target))
+		target.acid_act(acidpwr = 18, acid_volume = 25) // balanced
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/special/neurotoxin.dm]:
```suggestion
	else if(iscarbon(target))
		target.acid_act(acidpwr = 18, acid_volume = 25) // balanced
	else if(!isopenturf(target)
		target.acid_act(acidpwr = 50, acid_volume = 25) // does good damage to objects and structures
```

Comment (Moltijoe) [reply code/modules/projectiles/projectile/special/neurotoxin.dm]: no reason to restrict it so arbitrarily, all it needs to do is not be able to melt through the floor

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Comment by MajManatee** [comment]
local man doesnt know you can use an extingusher on acid

**Comment by iloveloopers** [comment]
this will probably needs buffs, it takes multiple shots to completely break a jumpsuit or door. but better weaker than overpowered

File: code/datums/mutations/fire_breath.dm [hunk 2]
```diff
@@ -93,3 +93,32 @@
...
+	cooldown_time = 18 SECONDS
+	spell_requirements = NONE
+	antimagic_flags = NONE
+	school = SCHOOL_EVOCATION
+	spell_requirements = NONE
+	sound = 'sound/effects/alien_spitacid.ogg'
+
+	active_msg = "You focus your acid spit!"
+	deactive_msg = "You relax."
+	projectile_type = /obj/item/projectile/bullet/acid
```
Comment (Moltijoe) [inline code/datums/mutations/fire_breath.dm]: what is this doing in the firebreath file

File: code/modules/projectiles/projectile/special/neurotoxin.dm [hunk 3]
```diff
@@ -12,3 +12,21 @@
...
+	speed = 1.8 // spit is not very fast
+
+obj/item/projectile/bullet/acid/on_hit(atom/target, blocked = FALSE)
+	if(isalien(target)) // shouldn't work on xenos
+		nodamage = TRUE
+	else if(iscarbon(target))
+		target.acid_act(acidpwr = 18, acid_volume = 25) // balanced
+	else if(!isopenturf(target))
+		target.acid_act(acidpwr = 50, acid_volume = 25) // does good damage to objects and structures
+	return ..()
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/special/neurotoxin.dm]: why is this with the neurotoxin projectile rather than the acid spit projectile

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — CHANGES_REQUESTED** [review]
undocumented sound addition

File: code/modules/projectiles/projectile/special/neurotoxin.dm [hunk 4]
```diff
@@ -12,3 +12,21 @@
...
+	range = 7
+	speed = 1.8 // spit is not very fast
+
+obj/item/projectile/bullet/acid/on_hit(atom/target, blocked = FALSE)
+	if(isalien(target)) // shouldn't work on xenos
+		nodamage = TRUE
+	else if(iscarbon(target))
+		target.acid_act(acidpwr = 18, acid_volume = 25) // balanced
+	else if(!isopenturf(target))
+		target.acid_act(acidpwr = 50, acid_volume = 25) // does good damage to objects and structures
```
Comment (Moltijoe) [inline code/modules/projectiles/projectile/special/neurotoxin.dm]:
```suggestion
		target.acid_act( 18, 25) // balanced
	else if(!isopenturf(target))
		target.acid_act( 50, 25) // does good damage to objects and structures
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Comment by iloveloopers** [comment]
theres no way the code isn't good now

**Review by Moltijoe — APPROVED** [review]
good enough

**Comment by iloveloopers** [comment]
come on balance council approve my PR already!!!

**Comment by Moltijoe** [comment]
Balance council:
>Reduce stability to 50 and reduce acid volume to 15
also, add a check making sure the user's mouth isn't covered when using it

File: code/datums/mutations/acid_spit.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	projectile_type = /obj/item/projectile/bullet/acid
+
+/datum/action/cooldown/spell/pointed/projectile/acid_spit/can_cast_spell(feedback)
+	. = ..()
+	var/mob/living/carbon/spitter = owner
+	if(!spitter.is_mouth_covered())
+		return
+	else
+		to_chat(spitter, span_notice("Something is covering your mouth!"))
+		return FALSE
```
Comment (Moltijoe) [inline code/datums/mutations/acid_spit.dm]:
```suggestion
	if(!get_location_accessible(owner, BODY_ZONE_PRECISE_MOUTH))
		to_chat(owner, span_notice("Something is covering your mouth!"))
		return FALSE
```

File: code/datums/mutations/acid_spit.dm [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	spell_requirements = NONE
+	sound = 'sound/effects/alien_spitacid.ogg'
+
+	active_msg = "You focus your acid spit!"
+	deactive_msg = "You relax."
+	projectile_type = /obj/item/projectile/bullet/acid
+
+/datum/action/cooldown/spell/pointed/projectile/acid_spit/can_cast_spell(feedback)
+	. = ..()
+	var/mob/living/carbon/spitter = owner
```
Comment (Moltijoe) [inline code/datums/mutations/acid_spit.dm]:
```suggestion
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — CHANGES_REQUESTED** [review]
please provide a screenshot or video of it working

File: code/datums/mutations/acid_spit.dm [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+	button_icon = 'icons/mob/actions/actions_xeno.dmi'
+	button_icon_state = "alien_neurotoxin_0"
+	active_icon_state = "alien_neurotoxin_1"
+	base_icon_state = "alien_neurotoxin_0"
+	active_overlay_icon_state = "bg_spell_border_active_red"
+	cooldown_time = 18 SECONDS
+	spell_requirements = NONE
+	antimagic_flags = NONE
+	school = SCHOOL_EVOCATION
+	spell_requirements = NONE
```
Comment (Moltijoe) [inline code/datums/mutations/acid_spit.dm]: you have this line twice
```suggestion
```

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Comment by iloveloopers** [comment]
https://github.com/yogstation13/Yogstation/assets/140007537/b944295b-6ba0-44a5-871a-c74cb8f815c3

**Review by Moltijoe — APPROVED** [review]

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**